modules/claims_api/app/services/claims_api/bgs_to_lighthouse_claims_mapper_service.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class BGSToLighthouseClaimsMapperService < ClaimsApi::Service
+    attr_accessor :bgs_claims, :lighthouse_claims
+
+    def initialize(bgs_claims:, lighthouse_claims:)
+      @bgs_claims        = bgs_claims
+      @lighthouse_claims = lighthouse_claims
+    end
+
+    def process
+      return [] unless claims_exist?
+
+      mapped_claims = map_claims
+      mapped_claims = append_remaining_lighthouse_claims(mapped_claims: mapped_claims) if lighthouse_claims.present?
+      mapped_claims
+    end
+
+    private
+
+    def claims_exist?
+      return true if bgs_claims.key?(:bnft_claim_detail) || lighthouse_claims.present?
+
+      false
+    end
+
+    def map_claims
+      bgs_claims[:bnft_claim_detail].map do |bgs_claim|
+        if find_bgs_claim_in_lighthouse_collection(claim: bgs_claim)
+          matching_claim = find_bgs_claim_in_lighthouse_collection(claim: bgs_claim)
+          remove_bgs_claim_from_lighthouse_collection(claim: matching_claim)
+          build_matched_claim(matching_claim: matching_claim, bgs_claim: bgs_claim)
+        else
+          build_unmatched_bgs_claim(bgs_claim: bgs_claim)
+        end
+      end
+    end
+
+    def find_bgs_claim_in_lighthouse_collection(claim:)
+      lighthouse_claims.find { |internal_claim| internal_claim.evss_id == claim[:bnft_claim_id] }
+    end
+
+    def remove_bgs_claim_from_lighthouse_collection(claim:)
+      lighthouse_claims.delete(claim)
+    end
+
+    def build_matched_claim(matching_claim:, bgs_claim:)
+      # this claim was submitted via Lighthouse, so use the 'id' the user is most likely to know
+      { id: matching_claim.id, type: bgs_claim[:bnft_claim_type_nm] }
+    end
+
+    def build_unmatched_bgs_claim(bgs_claim:)
+      { id: bgs_claim[:bnft_claim_id], type: bgs_claim[:bnft_claim_type_nm] }
+    end
+
+    def build_unmatched_lighthouse_claim(lighthouse_claim:)
+      { id: lighthouse_claim.id, type: lighthouse_claim.claim_type }
```

Comment:
i didnt think about this scenario... what's an example of a `bgs_claim[:bnft_claim_type_nm]` value? just want to make sure the value is the same structure (capitalized/lower case/camel case, etc.)

Reply:
ah, im guessing these are example values from BGS?
https://github.com/department-of-veterans-affairs/vets-api/pull/7395/files#diff-08f5bc438fee25554ed287b3f4ccd81a18e6b5f6eb4abdef35548b446a698b3fR429

if so, it looks like our claim types have the same structure at least

Reply:
`bnft_claim_type_nm: 'Appeals Control'` is an example value.